File: benchmark/objectivec/task8_http.m
#import <Foundation/Foundation.h>

void fetchURL(NSString* urlString, void (^completion)(BOOL success, NSString* body)) {
    NSURL* url = [NSURL URLWithString:urlString];
    NSURLSession* session = [NSURLSession sharedSession];

    [[session dataTaskWithURL:url completionHandler:^(NSData* data, NSURLResponse* response, NSError* error) {
        if (error) {
            completion(NO, error.localizedDescription);
        } else {
            completion(YES, [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding]);
        }
    }] resume];
}

File: benchmark/swift/task8_http.swift
import Foundation

func fetchURL(_ url: String) async throws -> String {
    guard let url = URL(string: url) else {
        throw URLError(.badURL)
    }
    let (data, _) = try await URLSession.shared.data(from: url)
    return String(data: data, encoding: .utf8) ?? ""
}
